lib/iex/lib/iex/info.ex

```diff
@@ -290,6 +290,32 @@ defimpl IEx.Info, for: Reference do
   end
 end
 
+defimpl IEx.Info, for: [Date, Time, NaiveDateTime] do
+  sigils = [{Date, "D"}, {Time, "T"}, {NaiveDateTime, "N"}]
+  @sigil sigils[@for]
```

Comment:
I think pattern matching here would be better:

```elixir
{sigil, repr} =
  case @for do
    Date -> {"D", "date"}
    ...
  end
```

You can use `unquote(sigil)` to inject it in (or store it in a module attribute later).